### ISSUE TITLE
Warning when try to call macros as function.

### ISSUE THREAD

**Issue by vkatsuba** [AdRoll/rebar3_hank#117]
Hi All,

When I try create macros and call it as function, after run `./rebar3 hank` I get warning about that macros is unused. Plugin gets from `main` branch.

**Warning:** `?MAKE_FUN_RUN is unused`

```erlang
%% ...

-ifdef(TEST).
-define(MAKE_FUN_RUN, ?MODULE:make_fun_run).
-else.
-define(MAKE_FUN_RUN, make_fun_run).
-endif.

%% ...

some_func() -> 
    ?MAKE_FUN_RUN(<<"Ping">>).

make_fun_run(Data) -> 
    io:format("~p~n", [Data]).
```

Regards,
--V

**Comment by elbrujohalcon** [comment]
Yeah… Those are the kinds of things that we're not ready to handle yet due to the limitations of `ktn_dodger`, `erl_syntax`, and `erl_parse`. It's the same thing that happens with the formatted.

In NextRoll… we use this kind of error as a **loud signal** that we should **not** write code like this.

Let me ask you (if you don't mind): Do you need this to happen this way because you want to mock `make_fun_run/1` for tests using `meck`?

**Comment by vkatsuba** [comment]
> Let me ask you (if you don't mind): Do you need this to happen this way because you want to mock make_fun_run/1 for tests using meck?

Yep and not only. But basically i didn't see here a big issue at all, I was just detect it and provide report about it. So, up to you. Do I need to close this issue or you will mark it as some `backlog` for the future code playing?

**Comment by elbrujohalcon** [comment]
I'll add it to the _Eventually_ milestone, yes. But I wanted to help you work around this issue in case you prefer not to add an ignore rule for this specific macro.
In any case, if you fancy… we can continue the conversation on how to design your tests not to require the macro trickery on Slack 😃

**Comment by vkatsuba** [comment]
Provided a possibly slightly dirty but workable approach.

**Comment by elbrujohalcon** [comment]
This is fixed in `0.4.1`: https://hex.pm/packages/rebar3_hank/0.4.1
